prog/vm/host_nexus.rb

```diff
@@ -206,11 +206,25 @@ def before_run
       hop_prep_reboot
     end
 
+    when_checkup_set? do
+      hop_unavailable if !available?
+    end
+
     Clog.emit("vm host utilization") { {vm_host_utilization: vm_host.values.slice(:location, :arch, :total_cores, :used_cores, :total_hugepages_1g, :used_hugepages_1g, :total_storage_gib, :available_storage_gib).merge({vms_count: vm_host.vms_dataset.count})} }
 
     nap 30
   end
 
+  label def unavailable
```

Comment:
Should we automatically update the allocation_state to "draining", or should the on-call engineer handle it?

Reply:
I prefer to leave this to on-call for a while, only to see failure modes and gain some understanding. I don't have any objection for updating the allocation_state (or some other ways of automated action) after that.